assert/utils.go

```diff
@@ -1350,6 +1350,58 @@ func formatEndsWithError(actual string, expected string, actualEndSufix string,
 	return ""
 }
 
+// findExactCaseMismatch finds the exact case mismatch for a substring within a string
+// Returns the position and the found substring if there's an exact case-only difference
+// If actualLower and substringLower are provided, they will be used to avoid recomputing
+func findExactCaseMismatch(actual, substring string, precomputedLower ...string) (int, string, bool) {
```

Comment:
Please consider passing a struct as argument

Relaying on a variadic of a fixed size that might be present or not is a bad thing

        if len(precomputedLower) >= 2 {